crates/runestick/src/function.rs

```diff
@@ -25,6 +26,139 @@ where
     inner: Inner<V>,
 }
 
+impl Function {
+    /// Perform an asynchronous call over the function which also implements
+    /// `Send`.
+    pub fn async_send_call<'a, A, T>(
+        &'a self,
+        args: A,
+    ) -> impl Future<Output = Result<T, VmError>> + Send + 'a
+    where
+        A: 'a + Send + Args,
+        T: 'a + Send + FromValue,
+    {
+        self.0.async_send_call(args)
+    }
```

Comment:
Hm. Isn't `Send` and `Sync` auto-implemented for the future if you do this?
```rust
    pub async fn async_send_call<A, T>(&self, args: A) -> Result<T, VmError>
    where
        A: 'a + Send + Args,
        T: 'a + Send + FromValue,
    {
        self.0.async_send_call(args).await
    }
```

Reply:
Yes; but that's a breaking API change -- current one shouldn't be. I think.

Reply:
Oh. Hm. I'm fairly confident that changing to a newtype already is a break. And I wouldn't worry too much about it at this stage. We just release a new breaking release on the next release.

I'm also not sure this strictly is breaking. I think the future generated is compatible with the signature you proposed.

Reply:
Yeah, true. Updated!